Sources/Networking/HTTPClient/ETagManager.swift

```diff
@@ -126,18 +140,24 @@ private extension ETagManager {
                                              response: HTTPResponse<Data?>,
                                              eTag: String) {
         if let data = response.body,
-           response.shouldStore(ignoreVerificationErrors: self.shouldIgnoreVerificationErrors),
-           let cacheKey = self.eTagDefaultCacheKey(for: request) {
-            let eTagAndResponse = Response(
-                eTag: eTag,
-                statusCode: response.statusCode,
-                data: data,
-                verificationResult: response.verificationResult
+           response.shouldStore(ignoreVerificationErrors: self.shouldIgnoreVerificationErrors) {
+            self.store(
+                Response(
+                    eTag: eTag,
+                    statusCode: response.statusCode,
+                    data: data,
+                    verificationResult: response.verificationResult
+                ),
+                for: request
             )
-            if let dataToStore = eTagAndResponse.asData() {
-                self.userDefaults.write {
-                    $0.set(dataToStore, forKey: cacheKey)
-                }
+        }
+    }
+
+    func store(_ response: Response, for request: URLRequest) {
+        if let cacheKey = self.eTagDefaultCacheKey(for: request),
+           let dataToStore = response.asData() {
```

Comment:
this function might no-op, I'm going to rename it so it's clear from the callsite that it's not guaranteed to actually store